src/ImageSharp/PixelFormats/PorterDuffFunctions.cs

```diff
@@ -0,0 +1,274 @@
+﻿// <copyright file="PorterDuffFunctions.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.PixelFormats
+{
+    using System.Numerics;
+    using System.Runtime.CompilerServices;
+
+    /// <summary>
+    /// Collection of Porter Duff alpha blending functions
+    /// </summary>
+    /// <typeparam name="TPixel">Pixel Format</typeparam>
+    /// <remarks>
+    /// These functions are designed to be a general solution for all color cases,
+    /// that is, they take in account the alpha value of both the backdrop
+    /// and source, and there's no need to alpha-premultiply neither the backdrop
+    /// nor the source.
+    /// Note there are faster functions for when the backdrop color is known
+    /// to be opaque
+    /// </remarks>
+    internal static class PorterDuffFunctions<TPixel>
+        where TPixel : IPixel
+    {
+        /// <summary>
+        /// Source over backdrop
+        /// </summary>
+        /// <param name="backdrop">Backgrop color</param>
+        /// <param name="source">Source color</param>
+        /// <param name="opacity">Opacity applied to Source Alpha</param>
+        /// <returns>Output color</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static TPixel NormalBlendFunction(TPixel backdrop, TPixel source, float opacity)

```

Comment:
I think it might be worth to provide `Vector4` versions for these functions.
Advantages:
- We can introduce a `BufferSpan<Vector4>`-based bulk-composer method in pixel blender.
- We can optimize `PixelBlender.Compose(BufferSpan<T>, BufferSpan<T>, float)` to use `PixelOperations<T>.ToVector4()` / `PixelOperations<T>.FromVector4()`.
- We could even use `Vector4` based bulk-composer in our processors. (See my comment on `PixelBlender<T>`)